src/transformers/models/trillsson_efficient/modeling_trillsson_efficient.py

```diff
@@ -0,0 +1,686 @@
+    for variable in init_vars.variables:
+        logger.info(f"Loading TF weight {variable.name} with shape {variable.shape}")
+        tf_weights[variable.name] = variable.numpy()
+    current_block = -1
+    current_block_index = ""
+    current_block_type = ""
+    name = ""
+    for vars_name in list(tf_weights.keys()):
+        m_name = vars_name.split("_", 1)  # max split = 1
+        pointer = model
+        array = tf_weights[vars_name]
+        for block_type in ["stem", "top", "block", "dense"]:
+            if block_type in m_name[0]:
+                current_block_type = block_type
+        if current_block_type != "dense":
+            pointer = getattr(pointer, current_block_type)
+            if current_block_type == "block":
+                block_index = m_name[0][-2:]
+                if block_index != current_block_index:
+                    current_block_index = block_index
+                    current_block += 1
+                pointer = pointer[current_block]
+                scope_names = m_name[1].split("/")
+            else:
+                scope_names = vars_name.split("/")
+            if "se" in scope_names[0]:
+                pointer = getattr(pointer, "se")
+            for name in scope_names:
+                if name in ["kernel:0", "gamma:0", "depthwise_kernel:0"]:
+                    pointer = getattr(pointer, "weight")
+                elif name in ["beta:0", "bias:0"]:
+                    pointer = getattr(pointer, "bias")
+                elif name == "moving_mean:0":
+                    pointer = getattr(pointer, "running_mean")
+                elif name == "moving_variance:0":
+                    pointer = getattr(pointer, "running_var")
+                else:
+                    pointer = getattr(pointer, name)
+        else:
+            pointer = getattr(pointer, "dense")
+            name = m_name[0].split("/")[1]
+            if name == "kernel:0":
+                pointer = getattr(pointer, "weight")
+            else:
+                pointer = getattr(pointer, "bias")
+
+        if name == "depthwise_kernel:0":
+            array = np.transpose(array, (3, 2, 0, 1))
+            array = np.transpose(array, (1, 0, 2, 3))
+        elif "kernel:0" in name:
+            if len(pointer.shape) == 2:  # copying into linear layer
+                array = array.squeeze().transpose()
+            else:
+                array = np.transpose(array, (3, 2, 0, 1))
+
+        try:
+            assert (
+                pointer.shape == array.shape
+            ), f"Pointer shape {pointer.shape} and array shape {array.shape} mismatched"
+        except AssertionError as e:
+            e.args += (pointer.shape, array.shape)
+            raise
+
+        pointer.data = torch.from_numpy(array.astype(np.float32))
+        tf_weights.pop(vars_name, None)
+
+    logger.info(f"Weights not copied to PyTorch model: {', '.join(tf_weights.keys())}.")
+    return model
+
+
+class Conv2DSamePadding(nn.Conv2d):
```

Comment:
Lets add the `TrillssonEfficient` prefix before all the classes that are proper to this model!